add yubikey init offering

biox opened the pull request:
for #49

arcxio commented on pa:
```diff
@@ -106,6 +106,22 @@ git_add_and_commit() {
 }
 
+yubikey_age_init() {
+    prompt='age-plugin-yubikey detected: generate an identity/receipient w/ yubikey?'
```
feels a bit verbose, I think just something like 'setup with yubikey?' is sufficient. and the shorter prompt could be put directly into yn

arcxio commented on pa:
```diff
@@ -106,6 +106,22 @@ git_add_and_commit() {
+            yn "$prompt" && {
+                age-plugin-yubikey -i > "$identities_file" ||
+                    die 'failed to generate YubiKey identity file'
+                age-plugin-yubikey -l > "$recipients_file" ||
```
current style is a blank line after every 'die'

arcxio commented on pa:
```diff
@@ -106,6 +106,22 @@ git_add_and_commit() {
+                    die 'failed to generate YubiKey identity file'
+                age-plugin-yubikey -l > "$recipients_file" ||
+                    die 'failed to generate YubiKey recipients file'
+            	printf '%s\n' 'YubiKey identity and recipient files generated.'
```
is this neccessary? when pa generates keys with age-keygen it doesn't announce successful generation

arcxio commented on pa:
```diff
@@ -271,6 +287,8 @@ main() {
     glob "$command" '[ds]*' && [ ! -f "$name.age" ] &&
         die "password '$name' doesn't exist"
 
+    yubikey_age_init
```
it would be clearer without separation from main imo

arcxio commented:
also it would be great to use yubikey in contrib/pa-rekey as well to make a switch to yubikey trivial

arcxio commented on pa:
```diff
@@ -106,6 +106,22 @@ git_add_and_commit() {
+
+    if command -v age-plugin-yubikey >/dev/null 2>&1; then
+        [ ! -f "$identities_file" ] && [ ! -f "$recipients_file" ] && {
+            yn "$prompt" && {
```
should pa prompt at all? maybe it's safe to assume that it should use yubikey if possible

biox commented:
@arcxio - I addressed your feedback in d9628e1ee7c14bedfac330beb40d3b9b33f4fb75 - i'm in favor of the prompt, just so we don't surprise users who happen to have a yubikey plugged in, as rare as that might be.

biox commented:
sounds good to me - i'll see if i can get to it back when im on my desktop w/ the yubikey
